### PR TITLE
Create the search mechanism in the `search` app

### LINKED ISSUE #116: Create Search mechanism - Apartment oriented
The Search mechanism will be Apartment oriented.

Anyone will be able to search apartments with **temporary** preferences and get results accordingly.
By default the preferences will be filled arbitrarily.
The search preferences can be modified.

### PR DESCRIPTION
**Pull request by AmitAharoni**
# Description
In this PR the search mechanism was created in the `search` app.

The logged-in user will be able to search apartments with preferences and get results accordingly.
By default the preferences will be filled according to the logged-in user type:

Seeker - with the seeker preferences.
Owner - with the apartment qualities.
Admin - with arbitrary preferences.

NOTES:
The search mechanism is Apartment oriented.
The Search mechanism works only for logged-in users.
The search preferences are temporary.
The search preferences can be modified.

## form screenshots:
image
## results Screenshots:
image

## Additional Information
In the next PR's the 'results.html' file should be upgraded.
The filtered apartments will be shown in cards with a Yes\no interface.
will be created by @nadavsu.

## PR workflow
- created a form for the search mechanism
- added templates for 'search' app …
- created the search mechanism in `views`.
- created test for the 'search' app.

### Issues closed by this PR
fix: #116, #119.

### PR Dependencies
None.

### CODE REVIEW

File: search/views.py [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+        if request.user.is_owner is True:
+            search_form = SearchForm(
+                instance=request.user.apartment,
+                initial={
+                    'min_rent': request.user.apartment.rent - 500,
+                    'max_rent': request.user.apartment.rent + 500}
+            )
+        elif request.user.is_seeker is True:
+            search_form = SearchForm(instance=request.user.seeker)
+        else:
```
Comment (DanielMy777) [inline search/views.py:30]: Is there an option to not be an owner nor a seeker? Do you mean like an admin?

File: search/views.py [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+    return render(request, 'search/search.html', context)
+
+
+def get_filtered_apartments(form):
+    return Apartment.objects.filter(**{
+            'is_relevant': True,
+            'start_date__lte': form['start_date'].value(),
+            'city': form['city'].value(),
+            'rent__gte': form['min_rent'].value(),
+            'rent__lte': form['max_rent'].value(),
```
Comment (Micha546) [inline search/views.py:52]: What is `rent__gte` and `rent__lte`?

**Review by Micha546 — APPROVED** [review]
Looking Great! 🔥 🔥 🔥

File: search/views.py (hunk 1 above)
Comment (AmitAharoni) [reply search/views.py:30]: Yes, that's for the admin.
@DanielMy777

File: search/views.py (hunk 2 above)
Comment (AmitAharoni) [reply search/views.py:52]: It's something like `min_rent  <= rent <= max_rent`

gte - Greater than or equal to.
lte -  Less than or equal to.

Django QuerySet API reference.
https://docs.djangoproject.com/en/3.2/ref/models/querysets/

@Micha546

Comment (Micha546) [reply search/views.py:52]: Ok, Looking great!

**Review by nadavsu — APPROVED** [review]

**Review by DanielMy777 — APPROVED** [review]
LGTM

**Pull request merged**